src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -58,19 +51,19 @@
 
 (defn add-query-params
   "Add any query parameters specified in configuration to the rule."
-  [rule {:keys [query-params]}]
+  [rule {{:keys [query-params]} :match-request}]
   (reduce-kv rules/query-param rule query-params))
 
 (schema/defn config->rule :- rules/Rule
   "Given a rule expressed as a map in the configuration return a Rule suitable
-  for use in a list with the allowed? function."
+   for use in a list with the allowed? function.
+
+   This assumes the configuration has been validated via
+   `validate-auth-config-rule!`."
   [m]
-  (if (contains? m :path)
-    (-> m
-        build-rule
-        (add-acl m)
-        (add-query-params m))
-    (throw (Exception. "Invalid config - missing required `path` key"))))
```

Comment:
We should be doing all validation in `validate-auth-config-rule!`, so this doesn't make sense here. I added note on the docstring to mention this.